velox/functions/prestosql/aggregates/tests/AggregationTestBase.cpp

```diff
@@ -136,6 +136,45 @@ void AggregationTestBase::testAggregations(
     assertResults(queryBuilder);
   }
 
+  if (!groupingKeys.empty() && allowInputShuffle_) {
+    SCOPED_TRACE("Run partial + final with spilling");
+    PlanBuilder builder(pool());
+    makeSource(builder);
+
+    // Spilling needs at least 2 batches of input. Use round-robin
+    // repartitioning to split input into multiple batches.
+    core::PlanNodeId partialNodeId;
+    builder.localPartitionRoundRobin()
+        .partialAggregation(groupingKeys, aggregates)
+        .capturePlanNodeId(partialNodeId)
+        .localPartition(groupingKeys)
+        .finalAggregation();
+
+    if (!postAggregationProjections.empty()) {
+      builder.project(postAggregationProjections);
+    }
+
+    auto spillDirectory = exec::test::TempDirectoryPath::create();
+
+    AssertQueryBuilder queryBuilder(builder.planNode(), duckDbQueryRunner_);
+    queryBuilder.config(core::QueryConfig::kTestingSpillPct, "100")
+        .config(core::QueryConfig::kSpillEnabled, "true")
+        .config(core::QueryConfig::kAggregationSpillEnabled, "true")
+        .config(core::QueryConfig::kSpillPath, spillDirectory->path)
+        .maxDrivers(4);
+
+    auto task = assertResults(queryBuilder);
+
+    // Expect > 0 spilled bytes unless there was no input.
```

Comment:
Comment needs updating.

Why can't we keep the original > 0 check? Would it be possible to fix the tests that use non-deterministic seed instead of changing the check?

Reply:
I am wondering if some test generates rows that goes to the same partition then even with local partition we can't trigger spilling as there is only vector. I guess there is some flakiness in the past which is hard to reproduce might be due to this.